Neptune/Services/IService.cs

```diff
@@ -1,11 +1,31 @@
-﻿namespace Neptune
+﻿using System;
+using System.Net.Http;
+
+namespace Neptune
 {
     public interface IService { }
-    public interface IMessageService<IS>
+    public class MessageService<IS>
         where IS:IService
     {
-        R SendMessage<M, R>(M msg)
-            where M:IMessage<IS>
-            where R:IResponse<M>;
+        private HttpClient client;
+
+        public MessageService(HttpClient client)
+        {
+            this.client = client;
+        }
+
+        R GetRequest<M, R>(M msg)
+            where M : IGetRequest<IS>
+            where R : IResponse<M>
+        {
+            throw new NotImplementedException();
+        }
+
+        R PostRequest<M, R>(M msg)
+            where M : IPostRequest<IS>
+            where R : IResponse<M>
+        {
+            throw new NotImplementedException();
```

Comment:
@ynigoreyes This is where we will
1. Serialize the msg contents
2. Post it to the uri provided in the msg
3. Get a response
4. Deserialize the response into the generic R type
5. Return.

Reply:
got it